Docs: 12-factor tutorial updates (#2257)
canonical/charmcraft: docs/tutorial/kubernetes-charm-django.rst

This tutorial should take 90 minutes for you to complete.

If you're new to the charming world, Django apps are
specifically supported with a template to quickly generate a

> **rock** and a matching template to generate a **charm**.

Reviewer: Since we're adjusting this paragraph, should we make `rock` a link to the equivalent rockcraft tutorial?

Reviewer 2: In the past I've advised including links so early in the tutorial. Since the user will be splitting their attention between this page and the terminal, I think we should strive not to distract them with alternate documents unless essential. I think the next paragraph is sufficient context for the introduction.

Author: I agree with @medubelko about not including a link here. Thanks all for your feedback!

A rock is a special kind of OCI-compliant container image, while a
charm is a software operator for cloud operations that use the Juju

> orchestration engine. The result is Django apps that

Reviewer 2: Maybe the singular would lead the reader's thoughts toward the activity they're about to do?

```suggestion
orchestration engine. The combined result is a Django app that
```

Author: Sounds good to me -- will make the change everywhere

Verify that the ``name`` is ``django-hello-world``.

> The ``platforms`` must match the architecture of your host. Check

Reviewer 2: ```suggestion
The ``platforms`` key must match the architecture of your host. Check
```

the architecture of your system:

.. code-block:: bash

    dpkg --print-architecture

> Edit the ``platforms`` entry in ``rockcraft.yaml`` if required.

Reviewer 2: > entry

"key" would be more consistent and accurate, but I'm not too fussed.

Author: I fine with being consistent and using "key" here -- will make the change

    :end-before: [docs:add-juju-model-end]
    :dedent: 2

> Include a constraint to the Juju model to specify your architecture:

Reviewer 2: An alternative phrasing could be:

```suggestion
Constrain the Juju model to your architecture:
```
